### PR TITLE
switch to TensorDomain constructor to set allocation_domain

### PR DESCRIPTION
**Pull request by jjsjann123**
Existing code relies on setAllocationDomain to update the allocation domain after tensor creation in `convertInputRfactorsToRoots`. However the initial contiguity flag isn't consistent with the root domain, which triggers assert.

### CODE REVIEW

**Comment by jjsjann123** [comment]
!build

**Comment by jjsjann123** [comment]
!build

File: test/test_allocation_domain.cpp [hunk 1]
```diff
@@ -1168,4 +1168,31 @@ TEST_F(NVFuserTest, AllocationDomainContiguityForBroadcast) {
   testValidate(fusion, outputs, {t0}, {t1}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, AllocationDomainContiguityForExplicitBroadcast) {
```
Comment (zasdfgbnm) [inline test/test_allocation_domain.cpp:1171]: Is this related to your segmenter change?

**Review by zasdfgbnm — APPROVED** [review]

File: test/test_allocation_domain.cpp (hunk 1 above)
Comment (jjsjann123) [reply test/test_allocation_domain.cpp:1171]: Don't remember pushing any segmenter changes recently....

But this comes from a python example for allocation domain when I was playing with my wip branch. :laughing:

Comment (zasdfgbnm) [reply test/test_allocation_domain.cpp:1171]: > Don't remember pushing any segmenter changes recently....

Isn't this PR changing segmenter?

Comment (jacobhinkle) [reply test/test_allocation_domain.cpp:1171]: Sorry I got here late...

I think @zasdfgbnm means the above function `convertInputRfactorsToRoots` only applies at segmentation. So for example we could have a test like
```c++
auto tv1 = segment_set(tv0);
tv1->setAllocationDomain({alloc[1], alloc[0]});
auto tv2 = neg(tv1);
```
Then have check that after segmentation the input to the second segment expects transposed inputs.

Comment (jjsjann123) [reply test/test_allocation_domain.cpp:1171]: oh... sorry about this. I missed where the comment is at. :face_with_head_bandage: 

So the example here would trigger an assert on our existing code base. With the changes above the test passed with no problem.
In fact function `convertInputRfactorsToRoots` is called on all inputs even when no kernel segmentation happens.

File: csrc/fusion_segmenter.cpp [hunk 2]
```diff
@@ -1487,8 +1487,7 @@ void convertInputRfactorsToRoots(Fusion* fusion) {
     }
 
     NVF_ERROR(new_root_domain.size() == tv->domain()->contiguity().size());
-    auto new_td = IrBuilder::create<TensorDomain>(
-        new_root_domain, tv->domain()->contiguity());
```
Comment (jjsjann123) [inline csrc/fusion_segmenter.cpp:1491]: So this is where the issue happened.

For tensor with `allocation domain`, we expect contiguity flag to match that, but not with `root domain`. so this line would trigger an assert.
Even though in the code logic below, we are updating allocation domain with `new_td->setAllocationDomain(new_alloc_domain, new_td->contiguity());`

Switching to the new TensorDomain constructor allows us to set root domain and allocation domain at the same time to avoid the error.
